src/frontend/core/executor/impl/StreamingTriangleCountExecutor.cpp

```diff
@@ -103,48 +106,46 @@ void StreamingTriangleCountExecutor::execute() {
 long StreamingTriangleCountExecutor::getTriangleCount(int graphId, std::string host, int port,
                                                       int dataPort, int partitionId, std::string masterIP,
                                                       std::string runMode, StreamingSQLiteDBInterface streamingDB) {
-    NativeStoreTriangleResult oldResult;
-    NativeStoreTriangleResult newResult;
-    oldResult.localRelationCount = 1;
-    oldResult.centralRelationCount = 1;
-    oldResult.result = 0;
+    NativeStoreTriangleResult oldResult{1, 1, 0};
 
     if (runMode == "1") {
        oldResult = retrieveLocalValues(streamingDB, std::to_string(graphId),
                                        std::to_string(partitionId));
     }
-
     int sockfd;
-    char data[DATA_BUFFER_SIZE];
-    struct sockaddr_in serv_addr;
-    struct hostent *server;
+    if (localSocketMap.find(port) == localSocketMap.end()) {
+        struct sockaddr_in serv_addr;
+        struct hostent *server;
 
-    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+        localSocketMap[port] = socket(AF_INET, SOCK_STREAM, 0);
 
-    if (sockfd < 0) {
-        std::cerr << "Cannot accept connection" << std::endl;
-        return 0;
-    }
+        if (sockfd < 0) {
+            std::cerr << "Cannot accept connection" << std::endl;
```

Comment:
We should have a logger here rather than using `cerr`

Reply:
done

Reply:
Ok